src/trivyHelper.ts

```diff
@@ -0,0 +1,95 @@
+
+const stableTrivyVersion = "0.5.0";
+const trivyLatestReleaseUrl = "https://api.github.com/repos/aquasecurity/trivy/releases/latest";
+const trivyToolName = "trivy";
+
+export async function getTrivy(): Promise<string> {
+    const latestTrivyVersion = await getLatestTrivyVersion();
+
+    let cachedToolPath = toolCache.find(trivyToolName, latestTrivyVersion);
+    if (!cachedToolPath) {
+        let trivyDownloadPath;
+        const trivyDownloadUrl = getTrivyDownloadUrl(latestTrivyVersion);
+        const trivyDownloadDir = `${process.env['GITHUB_WORKSPACE']}/_temp/tools/trivy`;
+        console.log(util.format("Could not find trivy in cache, downloading from %s", trivyDownloadUrl));
+
+        try {
+            trivyDownloadPath = await toolCache.downloadTool(trivyDownloadUrl, trivyDownloadDir);
+        } catch (error) {
+            throw new Error(util.format("Failed to download trivy from %s", trivyDownloadUrl));
+        }
+
+        fs.chmodSync(trivyDownloadPath, "777");
+        const untarredTrivyPath = await toolCache.extractTar(trivyDownloadPath);
+        cachedToolPath = await toolCache.cacheDir(untarredTrivyPath, trivyToolName, latestTrivyVersion);
+    }
+
+    const trivyToolPath = findTrivy(cachedToolPath);
+    fs.chmodSync(trivyToolPath, "777");
+
+    return trivyToolPath;
+}
+
+async function getLatestTrivyVersion(): Promise<string> {
+    return toolCache.downloadTool(trivyLatestReleaseUrl).then((downloadPath) => {
+        const response = JSON.parse(fs.readFileSync(downloadPath, 'utf8').toString().trim());
+        if (!response.tag_name) {
+            return stableTrivyVersion;
+        }
+
+        return response.tag_name.substring(1);
+    }, (error) => {
+        core.debug(error);
+        core.warning(util.format("Failed to read latest trivy verison from %s. Using default stable version %s", trivyLatestReleaseUrl, stableTrivyVersion));
+        return stableTrivyVersion;
+    });
+}
+
+function getTrivyDownloadUrl(trivyVersion: string): string {
+    const curOS = os.type();
+    switch (curOS) {
+        case "Linux":
+            return util.format("https://github.com/aquasecurity/trivy/releases/download/v%s/trivy_%s_Linux-32bit.tar.gz", trivyVersion, trivyVersion);
+
+        case "Darwin":
+            return util.format("https://github.com/aquasecurity/trivy/releases/download/v%s/trivy_%s_macOS-32bit.tar.gz", trivyVersion, trivyVersion);
+
+        default:
+            throw new Error(util.format("Container scanning is not supported for %s currently", curOS));
+    }
+}
+
+function findTrivy(rootFolder: string): string {
+    fs.chmodSync(rootFolder, '777');
+    var filelist: string[] = [];
+    walkSync(rootFolder, filelist, trivyToolName);
+    if (!filelist) {
+        throw new Error(util.format("Trivy executable not found in path %s", rootFolder));
+    }
+    else {
+        return filelist[0];
+    }
+}
+
+var walkSync = function (dir: string, filelist: string[], fileToFind: any) {
```

Comment:
In case we are doing findDockle and findTrivy, we can move two methods to commonHelper: findTool and walkSync.

Reply:
Removed the functions.